src/Nethermind/Nethermind.Network/Enode.cs

```diff
@@ -34,9 +35,10 @@ public Enode(PublicKey nodeKey, IPAddress hostIp, int port)
         public Enode(string enodeString)
         {
             string[] enodeParts = enodeString.Split(':');
-            _nodeKey = new PublicKey(enodeParts[1].Split('@')[0].TrimStart('/'));
+            string[] enodeParts2 = enodeParts[1].Split('@');
+            _nodeKey = new PublicKey(enodeParts2[0].TrimStart('/'));
             Port = int.Parse(enodeParts[2]);
-            HostIp = IPAddress.Parse(enodeParts[1].Split('@')[1]);
+            HostIp = IPAddress.TryParse(enodeParts2[1], out var ip) ? ip : Dns.GetHostAddresses(enodeParts2[1]).First();
```

Comment:
how will the system behave if we cannot resolve DNS?

Reply:
It will throw on First() as the sequence would be empty. I was thinking about providing cleaner message here but thought it was not wort it. WDYT?

Reply:
Added exception handling